code/script.js

```diff
@@ -222,104 +273,206 @@ const generateBoard = () => {
+      keep = false;
+      filterCharacters();
+    }
   }
 }
 
-// It'll filter the characters array and redraw the game board.
+// Function to filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
   const { category, value } = currentQuestion
-  // Show the correct alert message for different categories
-  if (category === 'accessories') {
+  if (category === 'animal') {
     if (keep) {
-      alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
+      alertCorrect(
+        `Yes, the villager is a ${value} 👏! Keep all the ${value} villagers.`
       )
     } else {
-      alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
+      alertIncorrect(
+        `No, the villager is not a ${value} 👎! Remove all ${value} villagers.`
       )
     }
-  } else if (category === 'other') {
-    // Similar to the one above
-  } else {
+  } else if (category === 'personality') {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      alertCorrect(
+        `Yes, the villager is ${value} type 👏! Keep all the ${value} type villagers.`
+      )
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alertIncorrect(
+        `No, the villager is not ${value} type 👎! Remove all the ${value} type villagers.`
+      )
+    }
+  } else if (category === 'hobby') {
+    if (keep) {
+      alertCorrect(
+        `Yes, the villager likes ${value} 👏! Keep all the villagers that like ${value}`
+      )
+    } else {
+      alertIncorrect(
+        `No, the villager hates ${value} 👎! Remove all the villagers that like ${value}!`
+      )
+    }
+  } else if (category === 'zodiac') {
+    if (keep) {
+      alertCorrect(
+        `Yes, the villager is a ${value} 👏! Keep all the villagers that are ${value}.`
+      )
+    } else {
+      alertIncorrect(
+        `No, the villager obviously isn't a ${value} 👎! Remove all the villagers that are ${value}.`
+      )
     }
   }
+  if (keep === true) {
+    charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.category] === value)
+    generateBoard();
+  } else {
+    charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.category] !== value)
+    generateBoard();
+  }
+}
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+// Custom Alert Correct Guess
+const alertCorrect = (message) => {
+  correct.style.display = "grid";
+  correctText.innerText = message;
+  correctButton.addEventListener('click', () => {
```

Comment:
I was about to comment that you maybe could put the eventListeners on line 400 and 409 in the end with the others, but I realized that it probably needs to be in the function since it triggers the correct and incorrect pictures to display = "none"? If that is so, then good thinking!  😁